libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,209 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solRequire(_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (auto current = begin(_code); current != end(_code); ++current)
+	{
+		auto const& item = m_items.emplace_back(createAssemblyItemFromJSON(*current, m_sourceList));
+		if (item == Instruction::JUMPDEST)
+			solThrow(AssemblyImportException, "JUMPDEST instruction without a tag");
+		else if (item.type() == AssemblyItemType::Tag)
+			solRequire(
+				++current != end(_code) &&
+					createAssemblyItemFromJSON(*current, m_sourceList) == Instruction::JUMPDEST,
+				AssemblyImportException,
+				"JUMPDEST expected after tag."
+			);
```

Comment:
When the expression is complex, I think its' more readable to just use `if`s and `solThrow()`.

Also, I'd avoid modifying values you're using in the same expression. I mean, is `*current` passed to `createAssemblyItemFromJSON()` even guaranteed to be evaluated before `++current`? Even if it is, doing `++current` separately is better because it removes any doubts about that.

```suggestion
		else if (item.type() == AssemblyItemType::Tag)
		{
		    ++current;
		    if (current != end(_code) && createAssemblyItemFromJSON(*current, m_sourceList) == Instruction::JUMPDEST)
			    solThrow(AssemblyImportException, "JUMPDEST expected after tag.");
		}
```

Reply:
Stop it @cameel - I suggested this even and it's just fine as is.

Reply:
Haha, sorry, I did not recheck previous suggestions so I did not notice that. I still think that's not a great way to do it though. Code that relies on evaluation order like this is a bad idea IMO.

Reply:
Hmm.. but @cameel is also right. I will use @cameel's suggestion here.